### PR TITLE
Enhance Settings & Welcome Pages and MCP Installation Flow (AST-107883)

### PR DESCRIPTION
**Pull request by cx-rahul-pidde**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

> This PR introduces several improvements and fixes across the Visual Studio extension

- Settings Page:
      -  Refactored logic for scanner enable/disable and user preferences.
      -  Improved error handling and configuration management.
      - Added/updated UI elements for better user experience.

- Welcome Page:
    - Fixed icon resource loading using constants and correct namespaces.
    - Ensured robust handling of embedded resources (.ico/.png).
    - Improved UI theming and contrast for better accessibility.

- MCP Installation:
    - Refactored MCP install/uninstall/config logic for reliability.
    - Enhanced error handling and user feedback during installation.
    - Improved config file management and safe dictionary access.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxOneAssistSettingsModuleTests.cs [hunk 1]
```diff
@@ -0,0 +1,122 @@
+using ast_visual_studio_extension.CxPreferences;
+using System.Runtime.Serialization;
+using Xunit;
+
+namespace ast_visual_studio_extension_tests.cx_unit_tests.cx_extansion_test
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxOneAssistSettingsModuleTests.cs]: typo : extansion → extension

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesModuleTests.cs [hunk 2]
```diff
@@ -0,0 +1,158 @@
+using ast_visual_studio_extension.CxPreferences;
+using System.Runtime.Serialization;
+using Xunit;
+
+namespace ast_visual_studio_extension_tests.cx_unit_tests.cx_extansion_test
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesModuleTests.cs]: typo : extansion → extension

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
+using System;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Windows.Forms;
+using ast_visual_studio_extension.CxExtension.Utils;
+using ast_visual_studio_extension.CxPreferences;
+using Xunit;
+
+namespace ast_visual_studio_extension_tests.cx_unit_tests.cx_extansion_test
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests.cs]: typo : extansion → extension

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Async.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
+using ast_visual_studio_extension.CxPreferences;
+using System.Reflection;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace ast_visual_studio_extension_tests.cx_unit_tests.cx_extansion_test
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Async.cs]: typo : extansion → extension

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Auth.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
+using ast_visual_studio_extension.CxPreferences;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace ast_visual_studio_extension_tests.cx_unit_tests.cx_extansion_test
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Auth.cs]: typo : extansion → extension

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Events.cs [hunk 6]
```diff
@@ -0,0 +1,104 @@
+using ast_visual_studio_extension.CxPreferences;
+using System;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Windows.Forms;
+using Xunit;
+
+namespace ast_visual_studio_extension_tests.cx_unit_tests.cx_extansion_test
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Events.cs]: typo : extansion → extension

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/McpConfigManagerTests.cs [hunk 7]
```diff
@@ -0,0 +1,297 @@
+using ast_visual_studio_extension.CxPreferences.Configuration;
+using System;
+using System.IO;
+using System.Json;
+using Xunit;
+
+namespace ast_visual_studio_extension_tests.cx_unit_tests.cx_extansion_test
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/McpConfigManagerTests.cs]: typo : extansion → extension

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/McpInstallServiceAsyncTests.cs [hunk 8]
```diff
@@ -0,0 +1,148 @@
+using ast_visual_studio_extension.CxPreferences.Configuration;
+using ast_visual_studio_extension.CxWrapper.Models;
+using Moq;
+using System;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace ast_visual_studio_extension_tests.cx_unit_tests.cx_extansion_test
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/McpInstallServiceAsyncTests.cs]: typo : extansion → extension

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/McpInstallServiceTests.cs [hunk 9]
```diff
@@ -0,0 +1,191 @@
+using ast_visual_studio_extension.CxPreferences.Configuration;
+using ast_visual_studio_extension.CxWrapper.Models;
+using Moq;
+using System;
+using Xunit;
+
+namespace ast_visual_studio_extension_tests.cx_unit_tests.cx_extansion_test
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/McpInstallServiceTests.cs]: typo : extansion → extension

File: ast-visual-studio-extension/CxPreferences/CxOneAssistSettingsModule.cs [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+        public void ApplyUserPreferencesToRealtimeSettings()
+        {
+            AscaCheckBox = UserPrefAscaRealtime;
+            OssRealtimeCheckBox = UserPrefOssRealtime;
+            SecretDetectionRealtimeCheckBox = UserPrefSecretDetectionRealtime;
+            ContainersRealtimeCheckBox = UserPrefContainersRealtime;
+            IacRealtimeCheckBox = UserPrefIacRealtime;
+        }
+
+        public void EnableAllRealtimeScanners()
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension/CxPreferences/CxOneAssistSettingsModule.cs:102]: EnableAllRealtimeScanners and and DisableAllRealtimeScanners check 5 boolean properties that are structurally identical. dont we need to check with few more combinations of true and false.

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests.cs [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+        private static CxPreferencesModule CreateModule()
+        {
+            return (CxPreferencesModule)FormatterServices
+                .GetUninitializedObject(typeof(CxPreferencesModule));
+        }
+
+        [Fact]
+        public void AuthMessages_ShouldMatchCentralizedConstants()
+        {
+            Assert.Equal("Validating...", CxConstants.AUTH_VALIDATE_IN_PROGRESS);
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests.cs]: Optional : This test tests constants, not behaviour

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests.cs [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+                ResetPreferencesUiState();
+
+                var ui = CxPreferencesUI.GetInstance();
+                var module = CreateModule();
+                module.ApiKey = "abc123";
+                module.AdditionalParameters = string.Empty;
+
+                ui.Initialize(module);
+
+                var connectButton = GetPrivateField<Button>(ui, "button1");
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests.cs:56]: "button1" is an auto-generated — it's meaningless

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Async.cs [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+{
+    public class CxPreferencesUITests_Async
+    {
+        [Fact]
+        public async Task ValidateApiKeyAsync_WithEmptyApiKey_DoesNotAuthenticate()
+        {
+            var ui = CxPreferencesUI.GetInstance();
+            var tbApiKey = ui.GetType().GetField("tbApiKey", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(ui) as System.Windows.Forms.TextBox;
+            tbApiKey.Text = string.Empty;
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Async.cs:22]: Null-unsafe reflection chain will throw NullReferenceException, not a test failure

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Async.cs [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+            var method = ui.GetType().GetMethod("ValidateApiKeyAsync", BindingFlags.NonPublic | BindingFlags.Instance);
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Async.cs:25]: Where is ValidateApiKeyAsync avialable in same test file?

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Async.cs [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+            await (Task)method.Invoke(ui, new object[] { true });
+            Assert.False(CxPreferencesUI.IsAuthenticated());
+        }
+
+        [Fact]
+        public async Task CompleteAuthenticationSetupAsync_WithNullConfig_DoesNotThrow()
+        {
+            var ui = CxPreferencesUI.GetInstance();
+            var method = ui.GetType().GetMethod("CompleteAuthenticationSetupAsync", BindingFlags.NonPublic | BindingFlags.Instance);
+            await (Task)method.Invoke(ui, new object[] { null, false });
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Async.cs]: There is no assertion. "Does not throw" is only a valid test if the method is explicitly documented to swallow errors on null input. Without an assertion, this test passes vacuously

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Async.cs [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+        public async Task InitializeMcpModuleAsync_WithNullConfig_DoesNotThrow()
+        {
+            var ui = CxPreferencesUI.GetInstance();
+            var method = ui.GetType().GetMethod("InitializeMcpModuleAsync", BindingFlags.NonPublic | BindingFlags.Instance);
+            if (method != null)
+            {
+                // Only test if method exists
+                try
+                {
+                    await (Task)method.Invoke(ui, new object[] { null });
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Async.cs]: This is the most problematic test in the file. It has three escape hatches that guarantee it always passes:

If the method doesn't exist → skipped silently
If the method throws → swallowed silently
If the method succeeds → no assertion to verify anything

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Events.cs [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+            instanceField?.SetValue(null, null);
+
+            var authField = typeof(CxPreferencesUI).GetField("_isAuthenticated", BindingFlags.NonPublic | BindingFlags.Static);
+            authField?.SetValue(null, false);
+        }
+
+        [Fact]
+        public void OnApiKeyChange_ResetsAuthAndDisablesRestoreSession()
+        {
+            ResetUiState();
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Events.cs]: ResetUiState() is a static method but called manually in every test, is it not possible to call from constructor?

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/McpInstallServiceAsyncTests.cs [hunk 18]
```diff
@@ -0,0 +1,148 @@
...
+            // Should complete without throwing
+            try
+            {
+                var result = await service.InstallAsync(config, typeof(McpInstallServiceAsyncTests));
+                Assert.NotNull(result);
+            }
+            catch (Exception ex)
+            {
+                // Expected if network/API calls fail, but should be caught
+                Assert.True(ex is Exception);
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/McpInstallServiceAsyncTests.cs]: The catch block ensures the test always passes regardless of what goes wrong. This test provides zero value:

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/McpInstallServiceTests.cs [hunk 19]
```diff
@@ -0,0 +1,191 @@
...
+            foreach (var testCase in testCases)
+            {
+                try
+                {
+                    var result = (string)method.Invoke(null, new object[] { testCase });
+                    Assert.NotNull(result);
+                }
+                catch
+                {
+                    // Some edge cases may fail, which is acceptable
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/McpInstallServiceTests.cs]: Do we really need an empty catch?

File: ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs [hunk 20]
```diff
@@ -0,0 +1,204 @@
...
+using System.Json;
+using System.Text;
+using Newtonsoft.Json.Linq;
+
+namespace ast_visual_studio_extension.CxPreferences.Configuration
+{
+    internal class McpConfigManager
+    {
+        internal const string ServerName = "Checkmarx";
+        internal const string DefaultMcpUrl = "https://ast-master-components.dev.cxast.net/api/security-mcp/mcp";
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs:11]: Why .dev URL? URL as the production default is a significant risk.

File: ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs [hunk 21]
```diff
@@ -0,0 +1,204 @@
...
+            if (changed)
+            {
+                servers[ServerName] = desiredServer;
+                WriteConfig(configPath, root);
+            }
+
+            return changed;
+        }
+
+        public virtual bool RemoveCheckmarxServer(out string configPath)
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs]: Why public virtual?

File: ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs [hunk 22]
```diff
@@ -0,0 +1,204 @@
...
+            {
+                ["command"] = "npx",
+                ["args"] = new JsonArray
+                {
+                    "mcp-remote",
+                    mcpUrl,
+                    "--transport",
+                    "http-first",
+                    "--header",
+                    "Authorization:" + apiKey,
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs:86]: No Bearer  prefix is added? does not required? => Authorization: Bearer

File: ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs [hunk 23]
```diff
@@ -0,0 +1,204 @@
...
+            if (string.IsNullOrWhiteSpace(json))
+                return new JsonObject();
+
+            try
+            {
+                JsonValue parsed = JsonValue.Parse(StripJsonComments(json));
+                JsonObject root = parsed as JsonObject;
+                return root ?? new JsonObject();
+            }
+            catch
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs]: replace with catch (JsonException)

File: ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs [hunk 24]
```diff
@@ -0,0 +1,204 @@
+using System;
+using System.IO;
+using System.Json;
+using System.Text;
+using Newtonsoft.Json.Linq;
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs:4]: using System.Json;
using Newtonsoft.Json.Linq;

System.Json is used for reading/building the config object, Newtonsoft.Json is used only for pretty-printing in WriteConfig. This creates an unnecessary dual dependency.

Can't we have one?

File: ast-visual-studio-extension/CxPreferences/Configuration/McpInstallService.cs [hunk 25]
```diff
@@ -0,0 +1,218 @@
...
+            if (config == null || string.IsNullOrWhiteSpace(config.ApiKey))
+                return false;
+
+            try
+            {
+                var wrapper = new CxCLI.CxWrapper(config, ownerType ?? GetType());
+                wrapper.AuthValidate();
+                return IsTenantMcpEnabled(wrapper, out _);
+            }
+            catch
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension/CxPreferences/Configuration/McpInstallService.cs]: here it catches all the type of exception like OutOfMemoryException as well.

this is something we should avoid something like below


catch (Exception ex) when (ex is not OutOfMemoryException)
{
    return true; // fail-open, but log the exception
}

File: ast-visual-studio-extension/CxPreferences/Configuration/McpInstallService.cs [hunk 26]
```diff
@@ -0,0 +1,218 @@
...
+        /// </summary>
+        private static bool IsTenantMcpEnabled(CxCLI.CxWrapper wrapper, out bool checkFailed)
+        {
+            checkFailed = false;
+            try
+            {
+                return wrapper.AiMcpServerEnabled();
+            }
+            catch
+            {
```
Comment (cx-rakesh-kadu) [inline ast-visual-studio-extension/CxPreferences/Configuration/McpInstallService.cs:195]: same comment as given above for catch block

**Review by cx-rakesh-kadu — CHANGES_REQUESTED** [review]

File: ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs (hunk 22 above)
Comment (cx-rahul-pidde) [reply ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs:86]: Bearer is not needed here for mcp

File: ast-visual-studio-extension/CxPreferences/CxOneAssistSettingsModule.cs (hunk 10 above)
Comment (cx-rahul-pidde) [reply ast-visual-studio-extension/CxPreferences/CxOneAssistSettingsModule.cs:102]: Other than this checkbox, only cmbContainersTool remains that is already covered in same class

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests.cs (hunk 12 above)
Comment (cx-rahul-pidde) [reply ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests.cs:56]: In wpf Designer this will auto-generated with same name like button1 , Label5 etc.

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Async.cs (hunk 14 above)
Comment (cx-rahul-pidde) [reply ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/CxPreferencesUITests_Async.cs:25]: var method = ui.GetType().GetMethod("ValidateApiKeyAsync", BindingFlags.NonPublic | BindingFlags.Instance);   Method Name - ValidateApiKeyAsync

File: ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs (hunk 20 above)
Comment (cx-rahul-pidde) [reply ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs:11]: This is the functionality . If in api key we will not get base url then we are using this default URL. Same in other 2 plugins as well

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/McpInstallServiceTests.cs (hunk 19 above)
Comment (cx-rahul-pidde) [reply ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/McpInstallServiceTests.cs]: Removed

File: ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/McpInstallServiceAsyncTests.cs (hunk 18 above)
Comment (cx-rahul-pidde) [reply ast-visual-studio-extension-tests/cx-unit-tests/cx-extension-tests/McpInstallServiceAsyncTests.cs]: Removed Test

File: ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs (hunk 21 above)
Comment (cx-rahul-pidde) [reply ast-visual-studio-extension/CxPreferences/Configuration/McpConfigManager.cs]: virtual lets Moq create a "fake" version of the method for testing. Without it, Moq can't create the fake, so the test breaks.

**Review by cx-rakesh-kadu — APPROVED** [review]
